### ISSUE TITLE
Convert all root calls to pathlib

### ISSUE THREAD

**Issue by jameshalgren** [NOAA-OWP/t-route#78]
Anywhere we are using 
```
root = os.path.dirname(os.path.abspath(__file__))
```
 or any variant of that (i.e., `os.path.aabspath("")`), should be changed to the more robust 
```
import pathlib
root = pathlib.Path("../..").resolve()
```

... taking care to make sure that the proper relative path (e.g., `../..` vs. `../../../`) is resolved based on the script location.

**Comment by groutr** [comment]
@jameshalgren I think that `pathlib.Path(".")` will resolve to the current working directory (`os.getcwd()`), not the script location.
For the script location, `pathlib.Path(__file__).resolve()` works.

**Comment by jameshalgren** [comment]
Closed; superseded by #210
